src/components/ScmRepositoriesList/ScmRepositoriesList.tsx

```diff
@@ -0,0 +1,115 @@
+import { ScmRepositoryName } from 'components/ScmRepositoryName/ScmRepositoryName';
+import { ServiceContainerLoading } from 'components/ServiceContainers/ServiceContainerLoading';
+import { Toolbar } from 'components/Toolbar/Toolbar';
+import { ToolbarItem } from 'components/Toolbar/ToolbarItem';
+
+// keeping also not supported operations for testing purposes
+const filterOptions: IFilterOptions = {
+  filterAttributes: {
+    internalUrl: {
+      id: 'internalUrl',
+      title: 'Internal URL',
+      placeholder: 'string | !string | s?ring | st*ng',
+      operator: '=like=',
+    },
+    externalUrl: {
+      id: 'externalUrl',
+      title: 'External URL',
+      placeholder: 'string | !string | s?ring | st*ng',
+      operator: '=like=',
+    },
+  },
+};
+
+const sortOptions: ISortOptions = {
+  internalUrl: {
+    id: 'internalUrl',
+    title: 'Internal URL',
+    tableColumnIndex: 0,
+  },
+  externalUrl: {
+    id: 'externalUrl',
+    title: 'External URL',
+    tableColumnIndex: 1,
+  },
+};
+
+interface IScmRepositoriesListProps {
+  serviceContainerScmRepositories: IServiceContainer;
+  componentId: string;
+}
+
+/**
+ * Component displaying list of SCM Repositories.
+ *
+ * @param serviceContainerScmRepositories - Service Container for SCM Repositories
+ * @param componentId - Component ID
+ */
+export const ScmRepositoriesList = ({ serviceContainerScmRepositories, componentId }: IScmRepositoriesListProps) => {
+  const { getSortParams } = useSorting(sortOptions, componentId);
+
+  return (
+    <>
+      <Toolbar>
+        <ToolbarItem>
+          <Filtering filterOptions={filterOptions} componentId={componentId} />
+        </ToolbarItem>
+        <ToolbarItem>
+          <ProtectedComponent>
+            <ActionButton action={() => console.log('Not implemented yet!')}>Create SCM Repository</ActionButton>
+          </ProtectedComponent>
+        </ToolbarItem>
+      </Toolbar>
+
+      <ContentBox borderTop>
+        <ServiceContainerLoading {...serviceContainerScmRepositories} title={PageTitles.repositories}>
+          <TableComposable variant="compact">
+            <Thead>
+              {/**
+               * If column order is changed, the property tableColumnIndex (see sortOptions) has to be updated.
+               * Better solution can be implemented in the future.
+               */}
+              <Tr>
+                <Th>Name</Th>
+                <Th sort={getSortParams(sortOptions['internalUrl'].id)}>Internal SCM URL</Th>
+                <Th sort={getSortParams(sortOptions['externalUrl'].id)}>External SCM URL</Th>
+                <Th width={15}>Pre-build Sync</Th>
+              </Tr>
+            </Thead>
+            <Tbody>
+              {serviceContainerScmRepositories.data?.content.map((scmRepository: SCMRepository, rowIndex: number) => (
+                <Tr key={rowIndex}>
+                  <Td>{<ScmRepositoryName isLink scmRepository={scmRepository} />}</Td>
```

Comment:
![image](https://user-images.githubusercontent.com/17566891/225636405-eca2d446-80f6-4078-b7d4-7c5d11831dbd.png)

Reply:
All fixed except ScmRepositoryName component as we discussed in the group.

> for now I would prefer keep only ScmRepositoryLink(new one) and utils/entityNameGenerators.ts with generateScmRepositoryName

chathttps://chat.google.com/room/AAAArxlOMus/yqRe78f2_7M